src/ClientData/CallstackDataTest.cpp

```diff
@@ -200,4 +200,124 @@ TEST(CallstackData, FilterCallstackEventsBasedOnMajorityStartExcludesFunctionToS
                                                                          event4, event5, event6}));
 }
 
+TEST(CallstackData, ForEachCallstackEventOfTidInTimeRangeDiscretized) {
+  CallstackData callstack_data;
+
+  const uint64_t cs_id = 12;
+  CallstackInfo cs{{0x11, 0x10}, CallstackType::kComplete};
+  callstack_data.AddUniqueCallstack(cs_id, std::move(cs));
+
+  constexpr uint32_t kTid = 42;
+  constexpr uint32_t kAnotherTid = 43;
+
+  const uint64_t time1 = 142;
+  CallstackEvent event1{time1, cs_id, kTid};
+  callstack_data.AddCallstackEvent(event1);
+
+  const uint64_t time2 = 242;
+  CallstackEvent event2{time2, cs_id, kTid};
+  callstack_data.AddCallstackEvent(event2);
+
+  const uint64_t time3 = 342;
+  CallstackEvent event3{time3, cs_id, kAnotherTid};
+  callstack_data.AddCallstackEvent(event3);
+
+  const uint64_t time4 = 442;
+  CallstackEvent event4{time4, cs_id, kTid};
+  callstack_data.AddCallstackEvent(event4);
+
+  std::vector<CallstackEvent> callstack_list;
+  auto insert_callstack = [&](const CallstackEvent& event) { callstack_list.push_back(event); };
+  const uint32_t kResolution = 2000;
+  callstack_data.ForEachCallstackEventOfTidInTimeRangeDiscretized(kTid, 0, 1000, kResolution,
```

Comment:
0 and 1000 could be variables.

Reply:
I think this won't be needed with small focused test cases - these variables are fairly trivial and just set a time range

Reply:
Done